### PR TITLE
feat(spec-runner): seperate driver spec runner

### PR DESCRIPTION
**Pull request by tassja**
This removes a number of folders that are now located in the [driver spec runner repo](https://github.com/PlaceOS/driver-spec-runner) and the image in the docker compose is now pulled from [Dockerhub](https://hub.docker.com/repository/docker/placeos/drivers-spec) instead of locally.

Closes [PlaceOS/drivers#147](https://github.com/PlaceOS/drivers/issues/147)
Closes [PlaceOS/drivers#5](https://github.com/PlaceOS/drivers/issues/5)

### CODE REVIEW

**Review by caspiano — COMMENTED** [review]
Need to restore the following

- `shard.*`
- `src/build.cr`

File: shard.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: drivers
+version: 2.2.0
+
+# compile target
+targets:
+  test-harness:
+    main: src/app.cr
+  report:
+    main: src/report.cr
+
```
Comment (caspiano) [inline shard.yml]:
```suggestion
```

File: docker-compose.yml [hunk 2]
```diff
@@ -6,16 +6,29 @@ services:
...
+      - ./shard.override.yml:/shard.override.yml
+      - ./shard.lock:/shard.lock
+    working_dir: /
+    command: shards update && shards install --ignore-crystal-version
+    environment:
+      - REDIS_URL=redis://redis:6379
+      - TZ=$TZ
   drivers:
-    image: placeos/drivers-spec:crystal-latest
+    image: tasssja/drivers-spec:crystal-${CRYSTAL_VERSION:-1.0.0}
```
Comment (caspiano) [inline docker-compose.yml:23]:
```suggestion
    image: placeos/drivers-spec:crystal-${CRYSTAL_VERSION:-1.0.0}
```

File: report [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# when signal 2 (SIGINT) is received
+trap "trap_ctrlc" 2
+
+docker-compose pull
+
+docker-compose up
+
+exit_code="0"
+
+docker-compose exec placeos-drivers /app/src/report.cr
```
Comment (caspiano) [inline report:23]: The `report` binary is already inside the spec runner image

**Comment by caspiano** [comment]
Closes #5

File: docker-compose.yml [hunk 4]
```diff
@@ -6,31 +6,33 @@ services:
     hostname: redis
     environment:
       - TZ=$TZ
-
+  install-shards:
+    build: .
```
Comment (caspiano) [inline docker-compose.yml:10]: you should be able to use an existing container with crystal in it

File: report [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+# initialise trap to call trap_ctrlc function
+# when signal 2 (SIGINT) is received
+trap "trap_ctrlc" 2
+
+docker-compose pull
+
+docker-compose build
+
+exit_code="0"
```
Comment (caspiano) [inline report:21]: Doesn’t look like anything is happening here with the exit code

**Review by caspiano — COMMENTED** [review]
`SHARDS_CACHE_PATH` env might be able to be mounted into the install and placeos-drivers containers

File: report [hunk 6]
```diff
@@ -18,8 +18,6 @@ docker-compose pull
 
 docker-compose build
 
-exit_code="0"
```
Comment (caspiano) [inline report]: The reason this is done is that the exit code of the process run in the container is then forwarded to be the exit code of the script.

https://github.com/place-labs/staff-api/blob/master/test#L21-L27

File: docker-compose.yml [hunk 7]
```diff
@@ -7,7 +7,9 @@ services:
     environment:
       - TZ=$TZ
   install-shards:
-    build: .
+    image: crystallang/crystal:${CRYSTAL_VERSION:-1.0.0}-alpine
+    restart: on-failure
+    working_dir: /app/
     command: "shards install --ignore-crystal-version"
```
Comment (caspiano) [inline docker-compose.yml:13]:
```suggestion
    command: shards install --ignore-crystal-version
```

**Review by caspiano — APPROVED** [review]

**Review by stakach — APPROVED** [review]
LGTM

**Pull request merged**